lib/evaluation/handlers/limitDisclosureEvaluationHandler.ts

```diff
@@ -19,18 +20,39 @@ export class LimitDisclosureEvaluationHandler extends AbstractEvaluationHandler
 
   public handle(pd: PresentationDefinition, vcs: VerifiableCredential[]): void {
     pd.input_descriptors.forEach((inDesc: InputDescriptor, index: number) => {
-      if (inDesc.constraints?.fields && inDesc.constraints?.limit_disclosure === Optionality.Required) {
-        this.enforceLimitDisclosure(vcs, inDesc.constraints.fields, index);
+      if (
+        inDesc.constraints?.fields &&
+        (inDesc.constraints?.limit_disclosure === Optionality.Required ||
+          inDesc.constraints?.limit_disclosure === Optionality.Preferred)
+      ) {
+        this.enforceLimitDisclosure(vcs, inDesc.constraints, index);
       }
     });
   }
 
-  private enforceLimitDisclosure(verifiableCredential: VerifiableCredential[], fields: Field[], idIdx: number): void {
+  private limitDisclosureSupported(vc: VerifiableCredential, vcIdx: number, idIdx: number): boolean {
+    const limitDisclosureSignatures = LIMIT_DISCLOSURE_SIGNATURES;
+    if (!limitDisclosureSignatures?.includes(vc.proof.type)) {
+      this.createLimitDisclosureNotSupportedResult(idIdx, vcIdx);
+      return false;
+    }
+    return true;
+  }
+
+  private enforceLimitDisclosure(
+    verifiableCredential: VerifiableCredential[],
+    constraints: Constraints,
+    idIdx: number
+  ): void {
+    const fields = constraints?.fields as Field[];
+    const limitDisclosure = constraints.limit_disclosure as Optionality;
     verifiableCredential.forEach((vc, index) => {
-      const verifiableCredentialToSend = this.createVcWithRequiredFields(vc, fields, idIdx, index);
-      if (verifiableCredentialToSend) {
-        verifiableCredential[index] = verifiableCredentialToSend;
-        this.createSuccessResult(idIdx, `$[${index}]`);
+      if (this.limitDisclosureSupported(vc, index, idIdx)) {
```

Comment:
I wouldn't expect a method with enforcement in the name to have optionally whether selective disclosure will happen. That distinction should have been made before calling this method

Reply:
Fixed!